Assignment_2/hexadecimalConversion/HexCalc.java

```diff
@@ -0,0 +1,355 @@
+		if (check(firstnumber) && check(secondnumber)) {
+			if (firstnumber.length() < secondnumber.length()) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	
+	/**
+	 * this is isEqual method used to check if firstnumber == secondnumber
+	 * 
+	 * @param {String} firstnumber
+	 * @param {String} secondnumber
+	 * 
+	 * @return {boolean} value
+	 */
+	public boolean isEqual(String firstnumber, String secondnumber) {
+		boolean value = false;
+
+		if (check(firstnumber) && check(secondnumber)) {
+			int flag = 0;
+			if (firstnumber.length() == secondnumber.length()) {
+
+				for (int count = 0; count < firstnumber.length(); count++) {
+
+					if (firstnumber.charAt(count) != secondnumber.charAt(count)) {
+						value = false;
+						break;
+					} else {
+						if (flag == firstnumber.length() - 1) {
+							value = true;
+							break;
+						}
+						flag++;
+					}
+
+				}
+			}
+
+		}
+		return value;
+	}
+
+	
+	/**
+	 * this isGreater method is used to check is firstnumber > secondnumber or
+	 * not
+	 * 
+	 * @param {String} firstnumber
+	 * @param {String} secondnumber
+	 * 
+	 * @return {boolean} value
+	 */
+	public boolean isGreater(String firstnumber, String secondnumber) {
+		boolean value = false;
+		
+		if (check(firstnumber) && check(secondnumber)) {
+
+			if (firstnumber.length() == secondnumber.length()) {
+
+				for (int count = 0; count < firstnumber.length(); count++) {
+
+					if (firstnumber.charAt(count) < secondnumber.charAt(count)) {
+						value = false;
+						break;
+					} else if (firstnumber.charAt(count) > secondnumber
+							.charAt(count)) {
+						value = true;
+					}
+
+				}
+			} else if (isLengthgreater(firstnumber, secondnumber)) {
+				value = true;
+			}
+
+		}
+		return value;
+	}
+
+	
+	/**
+	 * 
+	 * isSmaller method is used to check if firstnumber < secondnumber or not .
+	 * 
+	 * @param {String} firstnumber
+	 * @param {String} secondnumber
+	 * 
+	 * @return {boolean} value
+	 */
+	public boolean isSmaller(String firstnumber, String secondnumber) {
+		boolean value = false;
+		
+		if (check(firstnumber) && check(secondnumber)) {
+
+			if (firstnumber.length() == secondnumber.length()) {
+
+				for (int count = 0; count < firstnumber.length(); count++) {
+
+					if (firstnumber.charAt(count) > secondnumber.charAt(count)) {
+						value = false;
+						break;
+					} else if (firstnumber.charAt(count) < secondnumber
+							.charAt(count)) {
+						value = true;
+					}
+
+				}
+			} else if (isLengthsmaller(firstnumber, secondnumber)) {
+				value = true;
+			}
+
+		}
+
+		return value;
+	}
+
+	
+	/**
+	 * getHex is a method created to convert DECIMAL NO. to HEXADECIMAL NO.
+	 * 
+	 * @param {integer} decimalnumber
+	 * 
+	 * @return {String} value
+	 */
+	public String getHex(int decimalnumber) {
+		int remainder;
+		String value = "";
+		
+		char digit[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A',
+				'B', 'C', 'D', 'E', 'F' };

```

Comment:
Can be defined as a constant as `final char digit[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' };`

Reply:
changed